src/js/core/methods/helpers/cardanoSignTxLegacy.js

```diff
@@ -0,0 +1,121 @@
+import cbor from 'borc';
+import CardanoAuxiliaryDataSupplement from '../../../types/trezor/protobuf';
+import {
+    Enum_CardanoTxAuxiliaryDataSupplementType as CardanoTxAuxiliaryDataSupplementType,
+    Enum_CardanoTxWitnessType as CardanoTxWitnessType,
+} from '../../../types/trezor/protobuf';
+import {CardanoSignedTxData} from '../../../types/networks/cardano'
+
+const SHELLEY_WITNESSES_KEY = 0;
+const BYRON_WITNESSES_KEY = 2;
+
+const CATALYST_REGISTRATION_SIGNATURE_ENTRY_KEY = 61285;
+const CATALYST_REGISTRATION_SIGNATURE_KEY = 1;
+
+type WitnessResult = {
+    type: number,
+    pub_key: string,
+    signature: string,
+    chain_code?: string,
+};
+
+export const toLegacyParams = params => {
```

Comment:
I am not sure this is the best approach. Perhaps we could keep most of the old code in a separate file and decide earlier which version to use. In this approach, I first create the parameters for the new request and then look at FW version. If it's older, I use `toLegacyParams` to transform the parameters into the old format. This results in less duplication and less legacy code, but if the "new" params change, this transformation will maybe need to change also. Opinions?

Reply:
hopefully we would be able to deprecate legacy mode as soon as the streamed one is out and people had a chance to update. If anything changes though, we would need to change whatever handles that legacy signing

I'd cover the transformation with flow types at least though so the linter "screams" if any data structure changes (I see there's currently the //@flow annotation at the top missing btw, so I believe this file is left out from type checking)